### PR TITLE
Transition Xsc to Xray

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

* Remove the "404" error when `XSC` is not avaliable
* Remove old unused API
* Add the new API URL's to the XSC service in its new location as an inner service in Xray

### CODE REVIEW

File: xsc/services/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const (
+	XraySuffix                        = "/xray/"
+	xscSuffix                         = "/xsc/"
+	apiV1Suffix                       = "api/v1"
+	XscInXraySuffix                   = apiV1Suffix + xscSuffix
+	MinXrayVersionXscTransitionToXray = "3.108.0"
+)
+
+// From Xray version 3.108.0, XSC is transitioning to Xray as inner service. This function will return the backward compatibility URL.
```
Comment (orz25) [inline xsc/services/utils/utils.go:19]: I think the last part of the comment should be :
This function will return compatible URL

File: xray/services/xsc/xsc.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package xsc
```
Comment (orz25) [inline xray/services/xsc/xsc.go:1]: I think we should add a comment explaining the difference between XscInnerService and XscServicesManager. It's a bit confusing

File: xray/services/scan.go [hunk 3]
```diff
@@ -186,42 +180,19 @@ func (ss *ScanService) GetScanGraphResults(scanId string, includeVulnerabilities
...
-	xscResponse := XscPostContextResponse{}
-	if err = json.Unmarshal(body, &xscResponse); err != nil {
-		return "", errorutils.CheckError(err)
-	}
-	return xscResponse.MultiScanId, err
-}
-
 // IsXscEnabled will try to get XSC version. If route is not available, user is not entitled for XSC.
-func (ss *ScanService) IsXscEnabled() (xsxVersion string, err error) {
+func (ss *ScanService) IsXscEnabled() (xscVersion string, err error) {
```
[the remark this answers is not in the document]
Comment (attiasas) [reply xray/services/scan.go:184]: it calls the version API, if we call it anywhere we will do redundant API calls.
I think we don't use it anymore, I will delete it as well.

File: xray/services/xsc/xsc.go (hunk 2 above)
Comment (attiasas) [reply xray/services/xsc/xsc.go:1]: added comment

File: xsc/services/utils/utils.go (hunk 1 above)
Comment (attiasas) [reply xsc/services/utils/utils.go:19]: done

**Pull request merged**